salt/master.py

```diff
@@ -1979,6 +1980,40 @@ def __init__(self, opts, key):
         self.masterapi = salt.daemons.masterapi.LocalFuncs(opts, key)
         self.channels = []
 
+    def _list_valid_minions(self, clear_load):
+        delimiter = clear_load.get("kwargs", {}).get("delimiter", DEFAULT_TARGET_DELIM)
+        _res = self.ckminions.check_minions(
+            clear_load["tgt"], clear_load.get("tgt_type", "glob"), delimiter
+        )
+        tgt_minions = _res["minions"]
+        valid_minions = set()
+        for v in clear_load["valid_tgt"]:
+            _res = self.ckminions.check_minions(
+                v, clear_load.get("valid_tgt_type", "glob"), delimiter
+            )
+            valid_minions.update(_res["minions"])
+        ret = {"valid_minions": list(valid_minions), "tgt_minions": tgt_minions}
+        event_ret = {
+            "fun": "sys.tgt_match_ret",
+            "tgt": clear_load["tgt"],
+            "tgt_type": "glob",
+            "arg": ret,
+            "user": "root",
+            "ret": "",
+            "kwargs": {"blerp": self.opts["id"]},
+            "jid": clear_load.get("jid"),
+        }
+
+        missing = []
+        jid = self._prep_jid(event_ret, {})
+        payload = self._prep_pub([], jid, event_ret, {}, missing)
+        self.event.fire_event(payload, tagify([jid, "tgt_return"]))
+        # self.event.fire_event(
+        #    {"Return Some Things": jid, 'valid_minions': list(valid_minions), 'target_minions': tgt_minions},
+        #    tagify([jid, 'tgt_return'], "syndic"),
+        # )
```

Comment:
**chore** probably should remove this comment. I'm still not entirely clear when events need "fun" and "ret". I *think* I could skip a `_prep_pub`. Also... 

**question** does prepping a jid here do anything silly? :thinking:

Reply:
done. No longer prepping jid, so not actually relevant now.